networking_nsxv3/api/rpc.py

```diff
@@ -231,69 +170,82 @@ def plugin(self):
-            context, security_group_id, tag_name)
+    def get_security_group(self, context, security_group_id):
+        id_rev = db.get_security_group_revision(context, security_group_id)
+        if id_rev:
+            return {
+                "id": id_rev[0],
+                "revision_number": id_rev[1],
+                "tags": db.get_security_group_tag(context, security_group_id),
+                "ports": db.get_port_id_by_sec_group_id(context, sec_group_id)
+            }
 
     @log_helpers.log_method_call
-    @oslo_messaging.expected_exceptions(exceptions.ObjectNotFound)
-    def get_qos(self, context, qos_id):
-        return db.get_qos(context, qos_id)
+    def get_rules_for_security_groups_id(self, context, security_group_id):
+        return db.get_rules_for_security_groups_id(context, security_group_id)
 
     @log_helpers.log_method_call
-    def get_qos_bwl_rules(self, context, qos_id):
-        return db.get_qos_bwl_rules(context, qos_id)
+    def get_security_group_members_ips(self, context, security_group_id):
+        return \
+            db.get_security_group_members_ips(context, security_group_id) + \
+            db.get_security_group_members_address_bindings_ips(context, security_group_id)
 
     @log_helpers.log_method_call
-    def get_qos_dscp_rules(self, context, qos_id):
-        return db.get_qos_dscp_rules(context, qos_id)
+    def get_security_groups_for_host(self, context, host, limit, offset):
+        return db.get_security_groups_for_host(context, host, limit, offset)
 
     @log_helpers.log_method_call
-    @oslo_messaging.expected_exceptions(exceptions.ObjectNotFound)
-    def get_port(self, context, port_id):
-        return db.get_port(context, port_id)
-
+    def get_remote_security_groups_for_host(self, context, host, limit, offset):
+        return db.get_remote_security_groups_for_host(context, host, limit, offset)
+    
     @log_helpers.log_method_call
-    def get_port_security_groups(self, context, port_id):
-        return db.get_port_security_groups(context, port_id)
+    def has_security_group_used_by_host(self, context, host, security_group_id):
+        return db.has_security_group_used_by_host(context, host, security_group_id)
 
     @log_helpers.log_method_call
-    def get_port_allowed_pairs(self, context, port_id):
-        return db.get_port_allowed_pairs(context, port_id)
+    def get_port(self, context, host, port_id):
+        port = db.get_port(context, host, port_id)
 
-    @log_helpers.log_method_call
-    def get_port_addresses(self, context, port_id):
-        return db.get_port_addresses(context, port_id)
+        # NSX-T does not support CIDR as port manual binding - skipping X/X
 
-    @log_helpers.log_method_call
-    def get_rules_for_security_groups_id(self, context, security_group_id):
-        return db.get_rules_for_security_groups_id(context, security_group_id)
+        for ip in db.get_port_addresses(context, port_id):
+            if "/" in ip:
```

Comment:
I am not sure, what you are trying to do here. Are you trying to filter out netmasks? What about "/32" (in ipv4)?
Can't you filter by checking for  the mac address to achieve the same?

Probably a topic for a later change, but I think any filtering should happen via a database query.

Reply:
NSX-T does not support CIDR as port manual binding - skipping X/X, but for /32 it makes sense.

- [ ] Remove /32 from CIDR and use the base IP

Reply:
This is strange change is taken over by a commit from me, here the explanation:

Initial implementation of policy api used the the addresses collected from all remote security group hosts directly, Whereas mgmt api used netaddr python module to collapse continuous addresses to address codes. We hit without the "collapse" the limit of ips per group in one region with one customer, so I adapted the code to use the same collapse code that mgmt api is using.

Funny thing now: NSXT handles cidr independently from pure up addresses (1.2.3.4 and 1.2.3.4/32 are two distinct ip addresses in the group, LOLWTF!) , so I added the strange check for /32 that leaves the existing host addresses instead introducing duplicates.

Reply:
@notandy I read this as I can remove "/" from here and relay on the code bellow, correct?

def cidr(self, os_cidrs):
        return [str(ip).replace("/32", "") for ip in netaddr.IPSet(os_cidrs).iter_cidrs()]

Reply:
True, should be fine

Reply:
I am not so happy about that. the function is named `cidr`, so returning a cidr does make sense.
I would suggest to add simply a function to filter out hosts, and then check for either no netmask, or host-netmask.
Ideally, we also should check for IPv6, or does NSX-T does not IPv6?

Reply:
@fwiesel, I will correct the function to use the proper name for its purpose. Note that `netaddr.IPSet(os_cidrs).iter_cidrs()` works also for IPv6 addresses. And correct with `/32` we are doing small optimisation for IPv4 only, will make this optimisation only if the address is IPv4.
def get_compacted_cidrs(self, os_cidrs):
        """
        Reduce number of CIDRs based on the netmask overlapping
        """